setters/builder.go

```diff
@@ -0,0 +1,54 @@
+package setters
```

Comment:
why not putting it in a separated package. But this has to be private. It should have any public interface. Therefore put it in a internal package. something like `internal/setter`.

Also i'd rather singularize the package name.

The only "public" interface should be.
```go
type SetterFactory interface {
	BuildSetter(reflect.StructField) Setter
}

type Setter interface {	
	Set(string, interface{}) error	
}
```

everything else should remain private

Reply:
And also if you separate this package you should embed specific test in it with